src/commands/synthetics/reporters/junit.ts

```diff
@@ -0,0 +1,280 @@
+import c from 'chalk'
+import {promises as fs} from 'fs'
+import {Writable} from 'stream'
+import {Builder} from 'xml2js'
+
+import {PollResult, Reporter, Step, Test, Vitals} from '../interfaces'
+import {RunTestCommand} from '../run-test'
+
+interface Stats {
+  allowfailures: number
+  assertions: number
+  errors: number
+  failures: number
+  skipped: number
+  tests: number
+  warnings: number
+}
+
+interface XMLRunProperties extends Stats {
+  name: string
+}
+
+interface XMLRun {
+  $: XMLRunProperties
+  testsuite: XMLSuite[]
+}
+
+interface XMLSuiteProperties extends Stats {
+  name: string
+  time: number
+  timestamp: number
+}
+
+interface XMLSuite {
+  $: XMLSuiteProperties
+  properties: {
+    property: {$: {name: string; value: any}}[]
+  }
+  testcase: XMLStep[]
+}
+
+interface XMLStepProperties extends Stats {
+  allow_failure: boolean
+  is_skipped: boolean
+  name: string
+  substep_public_id?: string
+  time: number
+  type: string
+  url: string
+}
+
+interface XMLStep {
+  $: XMLStepProperties
+  browser_error: {$: {name: string; type: string}; _: string}[]
+  error: {$: {type: 'assertion'}; _: string}[]
+  vitals?: {$: Vitals}[]
+  warning: {$: {type: string}; _: string}[]
+}
+
+interface XMLJSON {
+  testsuites: {
+    $: {name: string}
+    testsuite: XMLRun[]
+  }
+}
+
+const getDefaultStats = (): Stats => ({
+  allowfailures: 0,
+  assertions: 0,
+  errors: 0,
+  failures: 0,
+  skipped: 0,
+  tests: 0,
+  warnings: 0,
+})
+
+// Return the stats from a given object
+// based on getDefaultStats
+const getStats = (obj: any): Stats => {
+  const baseStats = getDefaultStats()
+  for (const entry of Object.entries(baseStats)) {
+    const [key] = entry as [keyof Stats, number]
+    baseStats[key] = obj[key] || baseStats[key]
+  }
```

Comment:
nitpick, but would this be a bit simpler? 
```suggestion
  for (const [key, value] of Object.entries(baseStats)) {
    baseStats[key] = value || baseStats[key]
  }
```

Reply:
I think I had to do this for the typing.
Let me double check.

Reply:
yeah don't sweat it, it is also ok this way

Reply:
Yep, typing issue.
![image](https://user-images.githubusercontent.com/597828/122022130-97dd1800-cdc6-11eb-8b2b-99002019ba3f.png)